pynamodb/models.py

```diff
@@ -474,35 +457,42 @@ def refresh(self, consistent_read: bool = False, settings: OperationSettings = O
             raise ValueError("Cannot refresh this item from the returned class: {}".format(stored_cls.__name__))
         self.deserialize(item_data)
 
-    def get_operation_kwargs_from_instance(
+    def get_update_kwargs_from_instance(
         self,
-        key: str = KEY,
-        actions: Optional[Sequence[Action]] = None,
+        actions: List[Action],
         condition: Optional[Condition] = None,
         return_values_on_condition_failure: Optional[str] = None,
     ) -> Dict[str, Any]:
-        is_update = actions is not None
-        is_delete = actions is None and key is KEY
-        args, save_kwargs = self._get_save_args(null_check=not is_update)
+        hk_value, rk_value = self._get_hash_range_key_serialized_values()
 
-        version_condition = self._handle_version_attribute(
-            serialized_attributes={} if is_delete else save_kwargs,
-            actions=actions
-        )
+        version_condition = self._handle_version_attribute(actions=actions)
         if version_condition is not None:
             condition &= version_condition
 
-        kwargs: Dict[str, Any] = dict(
-            key=key,
-            actions=actions,
-            condition=condition,
-            return_values_on_condition_failure=return_values_on_condition_failure
-        )
-        if not is_update:
-            kwargs.update(save_kwargs)
-        elif 'range_key' in save_kwargs:
-            kwargs['range_key'] = save_kwargs['range_key']
-        return self._get_connection().get_operation_kwargs(*args, **kwargs)
+        return self._get_connection().get_operation_kwargs(hk_value, range_key=rk_value, key=KEY, actions=actions, condition=condition, return_values_on_condition_failure=return_values_on_condition_failure)
```

Comment:
I've really been meaning to setup black to lint 😅

Reply:
So I'll withhold formatting cause there's no point.

Reply:
exactly